src/oid_array.rs

```diff
@@ -0,0 +1,80 @@
+//! Bindings to libgit2's raw git_strarray type
+
+use std::ops::{Range,Deref};
+
+use oid::Oid;
+use raw;
+use util::Binding;
+use std::slice;
+
+/// An oid array structure used by libgit2
+///
+/// Some apis return arrays of oids which originate from libgit2. This
+/// wrapper type behaves a little like `Vec<&Oid>` but does so without copying
+/// the underlying Oids until necessary.
+pub struct OidArray {
+    raw: raw::git_oidarray,
+}
+
+/// A forward iterator over the Oids of an array, casted to `&Oid`.
+pub struct Iter<'a> {
+    range: Range<usize>,
+    arr: &'a OidArray,
+}
+
+impl OidArray {
+    /// Returns None if i is out of bounds.
+    pub fn get(&self, i: usize) -> Option<Oid> {
+        if i < self.raw.count as usize {
+            Some(self[i])
+        } else {
+            None
+        }
+    }
+
+    /// Returns an iterator over the Oids contained within this array.
+    pub fn iter(&self) -> Iter {
+        Iter { range: 0..self.len(), arr: self }
+    }
+
+    /// Returns the number of strings in this array.
+    pub fn len(&self) -> usize { self.raw.count as usize }
+}
+
+impl Deref for OidArray {
+    type Target = [Oid];
+
+    fn deref(&self) -> &[Oid] {
+        unsafe {
+            slice::from_raw_parts(self.raw.ids as *const Oid, self.raw.count as usize)
```

Comment:
Could you add a `debug_assert_eq!` that the size of `Oid` is the same as pointee here? e.g.

```rust
debug_assert_eq!(mem::size_of::<Oid>(), mem::size_of_val(&*self.raw.ids))
```